backend/fms_core/viewsets/sample_next_step_by_study.py

```diff
@@ -62,6 +63,35 @@ def destroy(self, request, pk=None):
         else:
             return HttpResponseBadRequest(errors or f"Missing sample-next-step-by-study ID to delete.")
 
+    @action(detail=False, methods=["post"])
+    def destroy_list(self, request):
+        sample_ids = request.data.get("sample_ids", [])
+        study = request.data.get("study", None)
+        stepOrder = request.data.get("step_order", None)
+        if not sample_ids:
+            return HttpResponseBadRequest("No sample IDs provided.")
+        if not study:
+            return HttpResponseBadRequest("No study ID provided.")
+        if not stepOrder:
+            return HttpResponseBadRequest("No step order provided.")
+        queryset = self.get_queryset().filter(sample_next_step__sample__id__in=sample_ids, study=study, step_order__order=stepOrder)
+        values_list = queryset.values_list("sample_next_step__sample", "study", "step_order__order")
+        errors = []
+        removed = {}
+        try:
+            with transaction.atomic():
+                for sample_id, study_id, order in values_list:
+                    sample = Sample.objects.get(id=sample_id)
```

Comment:
could be very DB intensive ... might be worth batching the sample and study objects queries (stored in a dict). The study is likely to be the same over and over ... PostGreSQL does some caching but it is better not to rely on the DB caching.